### PR TITLE
Simplify with-google-analytics example

### PR DESCRIPTION
**Pull request by Josehower**
## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

First of all thanks for this amazing project and all the help you provide with these examples.

It seems there is code duplication in this example. After some tests locally seem to _document.js is not necessary for `gtag` to work properly.

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_app.js#L30-L34

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_document.js#L13-L17

I am aware of https://github.com/vercel/next.js/pull/40645 and I would like to ask @dave-hay, @SukkaW and @ijjk to consider this is still necessary. If so why then not move all content of the scripts from _app to _document?

In any case, I am open to adding here some comments explaining what is the reason for this code duplication if necessary.

<hr/>

Changes that come from  https://github.com/vercel/next.js/pull/43897

1. The `event` hashChangeComplete should be removed since `/home` and `/home/#section` is not new pageview, but just reference to the same page. 

If we go from /home to /home/#section (with a button click or a link for example) this shouldn't trigger a new page visit on `gtag`. 

For this reason, I think we should revert the changes from https://github.com/vercel/next.js/pull/36079. If there is a better argument of why this should stay I am also open to creating comments to clarify this on the example since I don't think should be the default behavior and not useful in most cases.

2. The `id="gtag-init"` was added with no context to the example from https://github.com/vercel/next.js/pull/29530

If there is a reason for this id in the script to existing I am open to adding a comment that clarifies this since in my experience is not necessary at all.

Edit: Batching with https://github.com/vercel/next.js/pull/43897 as recommended by https://github.com/vercel/next.js/pull/43897#issuecomment-1344635000

### CODE REVIEW

**Comment by Josehower** [comment]
Batching with https://github.com/vercel/next.js/pull/43897 as recommended by https://github.com/vercel/next.js/pull/43897#issuecomment-1344635000

**Comment by SukkaW** [comment]
> FYI to maintainers, there's a PR open that's trying to _add_ more to the _document.js. So it can be closed if this one is approved: #43838

@maxproske @Josehower I just checked #43838 and the PR is valid. Google Analytics **does** require a very lightweight inline script in the `<head>`. So `_document` **is required** for Google Analytics setup.

**Comment by karlhorky** [comment]
> Google Analytics **does** require a very lightweight inline script in the `<head>`. So `_document` **is required** for Google Analytics setup.

Ok so if `_document` is required, maybe the suggestion in the description from @Josehower makes sense, to avoid the duplication mentioned in the description:

> If [it's necessary to have `_document`] why then not move all content of the scripts from `_app` to `_document`?

If this is acceptable, the changes from https://github.com/vercel/next.js/pull/43838 could be also merged into this PR.

**Comment by SukkaW** [comment]
> Ok so if `_document` is required, maybe the suggestion in the description from Josehower makes sense, to avoid the duplication mentioned in the description:

> > If [it's necessary to have `_document`] why then not move all content of the scripts from `_app` to `_document`?

Agree.

@Josehower Would you mind changing your PR to only remove `gtag-init` inside `_app` instead? So after #43838 and this PR merge, Google Analytics should work properly.

**Comment by karlhorky** [comment]
> only remove `gtag-init` inside `_app` instead? So after #43838 and this PR merge, Google Analytics should work properly.

There would still be another change required - removing the duplicate code that is mentioned in the description of the PR:

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_app.js#L30-L34

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_document.js#L13-L17

**Comment by karlhorky** [comment]
Actually, I think the proposal by @Josehower was that all Google Analytics content be moved from `_app` to `_document`, so these parts as well:

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_app.js#L7-L18

https://github.com/vercel/next.js/blob/9d97a1e34a8a6e09eb127292c730d1a8df63ebb6/examples/with-google-analytics/pages/_app.js#L22-L41

**Comment by karlhorky** [comment]
> only remove `gtag-init` inside `_app` instead?

Also, one more thing, there is another change in this PR (because @maxproske suggested that the changes be batched - which probably created more confusion):

Screenshot 2022-12-10 at 12 10 24

**Comment by Josehower** [comment]
I manage to solve the issue with WebVitals reported  on https://github.com/vercel/next.js/issues/40410 without the need of _document.js  by adding the <script> on the <Head>

reproduction example of the fix: https://stackblitz.com/edit/nextjs-jyzaxg?file=pages/_app.js

This fix is also compliant with the suggestion done by @SukkaW on https://github.com/vercel/next.js/pull/43838#pullrequestreview-1212500419

to summarize the changes proposed on this PR.

1. Remove code duplication on _app.js and _document.js by removing _document.js
2. Removing  event `hashChangeComplete` since /home and /home/#section is not a new pageview but a reference to the same page and shouldn't be reported as different pages to google.
3. Removing id="gtag-init" that was added with no additional info to the example from https://github.com/vercel/next.js/pull/29530 and seems to be unnecesary

File: examples/with-google-analytics/pages/_app.js [hunk 1]
```diff
@@ -17,27 +18,31 @@ const App = ({ Component, pageProps }) => {
...
-        strategy="afterInteractive"
+      <script
         dangerouslySetInnerHTML={{
           __html: `
             window.dataLayer = window.dataLayer || [];
             function gtag(){dataLayer.push(arguments);}
             gtag('js', new Date());
 
+            // Disable any default-enabled advertising features (turn them on later when we get consent)
+            gtag('set', 'allow_google_signals', false);
```
Comment (SukkaW) [inline examples/with-google-analytics/pages/_app.js]: IMHO we shouldn't add this in the PR:

- It is not required to set up Google Analytics
- Anyone who follows Google Analytics' instruction can add this easily

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: examples/with-google-analytics/pages/_app.js (hunk 1 above)
Comment (Josehower) [reply examples/with-google-analytics/pages/_app.js]: This is not part of the PR anymore. ill update the description, sorry for that

Comment (Josehower) [reply examples/with-google-analytics/pages/_app.js]: https://github.com/vercel/next.js/pull/43894/files

Comment (Josehower) [reply examples/with-google-analytics/pages/_app.js]: This is the commit that removes this code 4 days ago

https://github.com/vercel/next.js/pull/43894/commits/581c0af2ae06c1200ec055a7196394e9ea89d65e

File: examples/with-google-analytics/pages/_app.js [hunk 2]
```diff
@@ -10,35 +11,33 @@ const App = ({ Component, pageProps }) => {
...
       router.events.off('routeChangeComplete', handleRouteChange)
-      router.events.off('hashChangeComplete', handleRouteChange)
     }
   }, [router.events])
 
   return (
     <>
+      <Head>
+        <script
+          dangerouslySetInnerHTML={{
```
Comment (OlegLustenko) [inline examples/with-google-analytics/pages/_app.js:23]: **question**: is this safe to add `dangerouslySetInnerHTML` here?

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**